launchdarkly-android-client-sdk/src/main/java/com/launchdarkly/sdk/android/ConnectivityManager.java

```diff
@@ -97,40 +95,10 @@ public void upsert(LDContext context, DataModel.Flag item) {
 
         @Override
         public void setStatus(ConnectionMode newConnectionMode, Throwable error) {
-            ConnectionMode oldConnectionMode = newConnectionMode == null ? null :
-                    connectionMode.getAndSet(newConnectionMode);
-            LDFailure failure = null;
-            if (error != null) {
-                if (error instanceof LDFailure) {
-                    failure = (LDFailure)error;
-                } else {
-                    failure = new LDFailure("Unknown failure", error, LDFailure.FailureType.UNKNOWN_ERROR);
-                }
-                lastFailure.set(failure);
-            }
-            boolean updated = false;
-            if (newConnectionMode != null && oldConnectionMode != newConnectionMode) {
-                if (failure == null && newConnectionMode.isConnectionActive()) {
-                    connectionInformation.setLastSuccessfulConnection(System.currentTimeMillis());
-                }
-                connectionInformation.setConnectionMode(newConnectionMode);
-                updated = true;
-            }
-            if (failure != null) {
-                connectionInformation.setLastFailedConnection(System.currentTimeMillis());
-                connectionInformation.setLastFailure(failure);
-                updated = true;
-            }
-            if (updated) {
-                try {
-                    saveConnectionInformation();
-                } catch (Exception ex) {
-                    LDUtil.logExceptionAtErrorLevel(logger, ex, "Error saving connection information");
-                }
-                updateStatusListeners(connectionInformation);
-                if (failure != null) {
-                    updateListenersOnFailure(failure);
-                }
+            if (error == null) {
+                updateConnectionInfoForSuccess(newConnectionMode);
+            } else {
+                updateConnectionInfoForError(newConnectionMode, error);
             }
```

Comment:
For reviewers: Majority of the removed code has been moved into `updateConnectionInfoForSuccess` and `updateConnectionInfoForError`.